src/inflators/physics-shape.ts

```diff
@@ -0,0 +1,139 @@
+import { addComponent } from "bitecs";
+import { PhysicsShape } from "../bit-components";
+import { HubsWorld } from "../app";
+import { CONSTANTS } from "three-ammo";
+
+export enum Shape {
+  BOX = 0,
+  CYLINDER,
+  SPHERE,
+  CAPSULE,
+  CONE,
+  HULL,
+  HACD,
+  VHACD,
+  MESH,
+  HEIGHTFIELD
+}
+
+export enum Fit {
+  ALL = 0,
+  MANUAL
+}
+
+export enum Axis {
+  X = 0,
+  Y,
+  Z
+}
+
+export type PhysicsShapeParams = {
+  type: Shape;
+  fit: Fit;
+  halfExtents: [number, number, number];
+  minHalfExtent: number;
+  maxHalfExtent: number;
+  sphereRadius: number;
+  cylinderAxis: Axis;
+  margin: number;
+  offset: [number, number, number];
+  orientation: [number, number, number, number];
+  heightfieldData: number[];
+  heightfieldDistance: number;
+  includeInvisible: boolean;
+};
+
+const DEFAULTS = {
+  type: Shape.HULL,
+  fit: Fit.ALL,
+  halfExtents: [1, 1, 1],
+  minHalfExtent: 0,
+  maxHalfExtent: Number.POSITIVE_INFINITY,
+  sphereRadius: NaN,
+  cylinderAxis: Axis.Y,
+  margin: 0.01,
+  offset: [0, 0, 0],
+  orientation: [0, 0, 0, 1],
+  heightfieldData: [],
+  heightfieldDistance: 1,
+  includeInvisible: false
+};
+
+export const PHYSICS_SHAPE_FLAGS = {
+  INCLUDE_INVISIBLE: 1 << 0
+};
+
+const getTypeString = (eid: number) => {
+  return Object.values(CONSTANTS.SHAPE)[PhysicsShape.type[eid]];
+};
+
+const getFitString = (eid: number) => {
+  return Object.values(CONSTANTS.FIT)[PhysicsShape.fit[eid]];
+};
+
+const getCylinderAxisString = (eid: number) => {
+  switch (PhysicsShape.cylinderAxis[eid]) {
+    case Axis.X:
+      return "x";
+    case Axis.Y:
+      return "y";
+    case Axis.Z:
+      return "z";
+  }
+};
+
+export const getShapeFromPhysicsShape = (eid: number) => {
```

Comment:
I guess we will need to also update the ammo-shape library if we want to do this.

Reply:
To do it well, probably. We can probably get a lot just changing the side of the postMessage bridge that lives on the main thread (the stuff in physics-system.js). Its just an extra copy of the state. Will be awhile till doing so makes sense though, since the aframe components would also need to be reworked unless we get rid of them first.